plugins/modules/cloud/univention/udm_dns_record.py

```diff
@@ -124,33 +132,44 @@ def main():
     changed = False
     diff = None
 
+    workzone = zone
+    workname = name
+    if type == 'ptr_record':
+        workzone = ('.').join(reversed(zone.split('.'))) + ".in-addr.arpa"
+        workname = ('.').join(reversed(name.split('.')[len(zone.split('.')):]))
```

Comment:
I don't know UDM or their API, but: does this also work with IPv6 addresses? These look somewhat different (see for example https://www.ionos.com/digitalguide/hosting/technical-matters/ptr-record/#c198217).

Reply:
(In any case, this is probably something that should be computed by code which knows how to handle these details, like https://docs.python.org/3/library/ipaddress.html#ipaddress.IPv4Address.reverse_pointer)

Reply:
You're right, the code works only for IPv4 addresses. My Univention server is v4 only. I'll check how to make it more generic. Currently it's in a "works for me" state.